### PR TITLE
Add Alternative Authentication Method.

### PR DESCRIPTION
**Pull request by jobannon**
# Summary 

Updates the client credentials grant information with the idea of authenticating with either 
- basic auth 
or
-  with a combined `client id` and `client secret` in a post to the `token` endpoint.

related:
https://fusionauth.io/community/forum/topic/1175/client-credentials-grant-flow-basic-auth-or-client_id-and-client_secret-in-the-body

### CODE REVIEW

File: site/docs/v1/tech/oauth/endpoints.adoc [hunk 1]
```diff
@@ -934,6 +934,17 @@ Ensure you are sending these parameters as form encoded data in the request body
...
 
 [.api]
+[field]#client_id# [type]#[String]# [optional]#Optional#::
+The unique client identifier.
+The client Id is the Id of the FusionAuth Entity in which you are attempting to authenticate.
++
+This parameter is optional when the `Authorization` header is provided.
+When the `Authorization` header is not provided, this parameter is then required.
+
+[field]#client_secret# [type]#[String]# [optional]#Optional# [since]#Available since 1.5.0#::
```
Comment (robotdan) [inline site/docs/v1/tech/oauth/endpoints.adoc:944]: The client credentials is new, so we don't need a version flag on this one.

**Comment by mooreds** [comment]
My main question was "is sending it in the request body compliant with RFC 6749?" It looks like it is, as defined in this section: https://datatracker.ietf.org/doc/html/rfc6749#section-2.3.1

File: site/docs/v1/tech/oauth/endpoints.adoc [hunk 2]
```diff
@@ -934,6 +934,17 @@ Ensure you are sending these parameters as form encoded data in the request body
 Do not send these parameters in a query string or as JSON.
 
 [.api]
+[field]#client_id# [type]#[String]# [optional]#Optional#::
+The unique client identifier.
+The client Id is the Id of the FusionAuth Entity in which you are attempting to authenticate.
++
+This parameter is optional when the `Authorization` header is provided.
+When the `Authorization` header is not provided, this parameter is then required.
```
Comment (mooreds) [inline site/docs/v1/tech/oauth/endpoints.adoc:942]:
```suggestion
When the `Authorization` header is not provided, this parameter is required.
```

File: site/docs/v1/tech/oauth/endpoints.adoc [hunk 3]
```diff
@@ -934,6 +934,17 @@ Ensure you are sending these parameters as form encoded data in the request body
...
+
+[field]#client_secret# [type]#[String]# [optional]#Optional# [since]#Available since 1.5.0#::
+The client secret of the Entity.
+This value may optionally be provided in the request body instead of the `Authorization` header documented above.
```
Comment (mooreds) [inline site/docs/v1/tech/oauth/endpoints.adoc:946]:
```suggestion
This parameter is optional when the `Authorization` header is provided.
When the `Authorization` header is not provided, this parameter is required.
```

Would re-use the same verbiage as used with `client_id`.

File: site/docs/v1/tech/oauth/endpoints.adoc [hunk 4]
```diff
@@ -1022,6 +1033,7 @@ The value will be `Bearer`.
 include::docs/v1/tech/oauth/_client_credentials_example_json_response.json[]
 ----
 
+
```
Comment (mooreds) [inline site/docs/v1/tech/oauth/endpoints.adoc:1036]: Nit, but not sure what this extra space buys us :)

File: site/docs/v1/tech/oauth/endpoints.adoc [hunk 5]
```diff
@@ -934,6 +934,17 @@ Ensure you are sending these parameters as form encoded data in the request body
 Do not send these parameters in a query string or as JSON.
```
Comment (mooreds) [inline site/docs/v1/tech/oauth/endpoints.adoc:934]: We should also probably modify the "You must provide client authentication using the Basic Authentication Scheme authorization header." verbiage in the request header section too.

**Review by mooreds — CHANGES_REQUESTED** [review]
A few changes requested.

**Pull request merged**